src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Pipeline/Steps/ExceptionEventsPipelineStep.cs

```diff
@@ -105,5 +110,46 @@ private static ReadOnlySpan<StackFrame> ComputeEffectiveCallStack(Exception exce
             }
             return ReadOnlySpan<StackFrame>.Empty;
         }
+
+        private ulong GetExceptionId(Exception? exception)
+        {
+            if (null == exception)
+                return 0;
+
+            ulong? exceptionId = null;
+            if (exception.Data.Contains(ExceptionIdKey))
+            {
+                exceptionId = (ulong?)exception.Data[ExceptionIdKey];
+            }
+
+            if (!exceptionId.HasValue)
+            {
+                lock (exception.Data)
+                {
+                    if (!exceptionId.HasValue)
```

Comment:
Is this intentionally checking against the same thing as L125?

Reply:
This is wrong, I'll go fix it. It needs to get the value from the Data property again.